### ISSUE TITLE
Aliased symbol also exported under non-aliased name.

### ISSUE THREAD

**Issue by mbostock** [rollup/rollup#438]
index.js:

```js
export {
  default as axis,
  orientBottom as axisOrientBottom
} from "./src/axis";
```

src/axis.js:

```js
export var orientBottom = {name: "bottom"};

export default function() {
  var orient;

  function axis() {}

  axis.orient = function() {
    orient = orientBottom;
  };

  return axis;
};
```

Resulting Rollup with `-f amd`:

```js
define(['exports'], function (exports) { 'use strict';

  exports.orientBottom = {name: "bottom"};

  function axis() {
    var orient;

    function axis() {}

    axis.orient = function() {
      orient = exports.orientBottom;
    };

    return axis;
  };

  exports.axis = axis;
  exports.axisOrientBottom = exports.orientBottom;

});
```

Note that the bottom orientation is unexpectedly exported twice: once as `orientBottom` (wrongly) and once as `axisOrientBottom` (correctly).

**Comment by mbostock** [comment]
Here’s an even simpler case for src/axis.js:

```js
export var orientBottom;

export default function() {
  var orient;
  orient = orientBottom;
};
```

Which produces:

```js
define(['exports'], function (exports) { 'use strict';

  exports.orientBottom;

  function axis() {
    var orient;
    orient = exports.orientBottom;
  };

  exports.axis = axis;
  exports.axisOrientBottom = exports.orientBottom;

});
```

**Comment by mbostock** [comment]
I can work around this issue by creating a var instead of an alias in index.js:

```js
import {orientBottom} from "./src/axis";
export {default as axis} from "./src/axis";
export var axisOrientBottom = orientBottom;
```

I assume this has something to do with exports being bindings rather than constants. Like, it’s protecting against someone importing axisOrientBottom and the setting the value to something else, yeah?

**Comment by mbostock** [comment]
Another interesting observation is that it produces broken code if the exported symbol is a function. In src/axis.js:

```js
export function orientBottom() {};

export default function() {
  var orient;
  orient = orientBottom;
};
```

Which produces:

```js
define(['exports'], function (exports) { 'use strict';

  function exports.orientBottom() {};

  function axis() {
    var orient;
    orient = exports.orientBottom;
  };

  exports.axis = axis;
  exports.axisOrientBottom = exports.orientBottom;

});
```

Which then fails because “exports.orientBottom” isn’t a valid function name.

**Comment by mbostock** [comment]
Although as I understand it (per [2ality](http://www.2ality.com/2015/07/es6-module-exports.html)), the imported ES6 binding should be immutable: no one should be able to reassign the `axisOrientBottom` symbol outside of the module that defines it.

That can’t be strictly enforced by Rollup when converting to ES5 (unless you wanted to protect the `exports` with a getter/setter rather than using a dumb object, or using Object.freeze, neither of which I’d prefer due to compatibility reasons). But I don’t really care about preventing people from writing code. :)

**Comment by Rich-Harris** [comment]
Fixed at last, in 0.25.1 (the first bug, not the `function exports.whatever` one – tracking that in #454). Thanks
